### PR TITLE
[HOTFIX] Rewrote address splitting function to work with bad addresses

### PR DESCRIPTION
**Pull request by SparK-Cruz**
Bad addresses with missing numbers broke the checkout page.

### CODE REVIEW

File: woocommerce-gateway-ebanx/includes/class-wc-ebanx-gateway-utils.php [hunk 1]
```diff
@@ -141,65 +141,39 @@
 	);
 
 	/**
-	 * This function splits an address line like for example "Pallaswiesenstr. 45 App 231" into its individual parts.
-	 * Supported parts are additionToAddress1, streetName, houseNumber and additionToAddress2. AdditionToAddress1
-	 * and additionToAddress2 contain additional information that is given at the start and the end of the string, respectively.
-	 * Unit tests for testing the regular expression that this function uses exist over at https://regex101.com/r/vO5fY7/1.
-	 * More information on this functionality can be found at http://blog.viison.com/post/115849166487/shopware-5-from-a-technical-point-of-view#address-splitting.
+	 * Fatia um endereço em nome da rua, número e complemento
```
Comment (cezarlz) [inline woocommerce-gateway-ebanx/includes/class-wc-ebanx-gateway-utils.php]: Comments are only in English :)

File: woocommerce-gateway-ebanx/includes/class-wc-ebanx-gateway-utils.php [hunk 2]
```diff
@@ -141,65 +141,39 @@
...
 	 *
-	 * @param string $address
+	 * @param  string $address Endereço a ser fatiado
```
Comment (cezarlz) [inline woocommerce-gateway-ebanx/includes/class-wc-ebanx-gateway-utils.php]: Same above.

File: woocommerce-gateway-ebanx/includes/class-wc-ebanx-gateway-utils.php [hunk 3]
```diff
@@ -141,65 +141,39 @@
...
-		   \s*[\/,]?\s*(?:\sNo[.:])?\s*
-			   (?P<B_House_number>\pN+\s*-?[a-zA-Z]?(?:\s*[-\/\pP]?\s*\pN+(?:\s*[\-a-zA-Z])?)*|
-			   [IVXLCDM]+(?!.*\b\pN+\b))(?<!\s) # House number
-		   \s*(?:(?:[,\/]|(?=\#)|\s)\s*(?!\s*No\.)\s*
-			   (?P<B_Addition_to_address_2>(?!\s).*?))? # Addition to address 2
-		   )
-		   \s*\Z/xu';
-
-		$result = preg_match($regex, $address, $matches);
+	public static function split_street($address) {
```
Comment (cezarlz) [inline woocommerce-gateway-ebanx/includes/class-wc-ebanx-gateway-utils.php]: Are you really right about this? It's a huge difference between the regexs... Why this?

File: woocommerce-gateway-ebanx/includes/class-wc-ebanx-gateway-utils.php [hunk 4]
```diff
@@ -141,65 +141,39 @@
...
+		$result = preg_match('/^([^,\-\/\#0-9]*)\s*[,\-\/\#]?\s*([0-9]+)\s*[,\-\/]?\s*([^,\-\/]*)(\s*[,\-\/]?\s*)([^,\-\/]*)$/', $address, $matches);
 
-		if ($result === 0) {
-			throw new \InvalidArgumentException(sprintf('Address \'%s\' could not be splitted into street name and house number', $address));
-		} elseif ($result === false) {
-			throw new \RuntimeException(sprintf('Error occurred while trying to split address \'%s\'', $address));
+		if ($result === false) {
+			throw new \RuntimeException(sprintf('Problemas ao interpretar o endereço: \'%s\'', $address));
```
Comment (cezarlz) [inline woocommerce-gateway-ebanx/includes/class-wc-ebanx-gateway-utils.php]: The string is in pt-br, please, put in en-us.

File: woocommerce-gateway-ebanx/includes/class-wc-ebanx-gateway-utils.php (hunk 3 above)
Comment (SparK-Cruz) [inline woocommerce-gateway-ebanx/includes/class-wc-ebanx-gateway-utils.php]: That regex was not working when address had no number... a merchant added an address number field to their checkout (using checkout manager) and that broke things as address had no number and regex simply failed...
As it was too complex we rewrote the regex to be simplier and to match with weird addresses anyway.

**Pull request merged**